addons/mrp_subcontracting_dropshipping/models/purchase.py

```diff
@@ -1,13 +1,43 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models, _
+from odoo import models, _, fields, api
 from odoo.exceptions import UserError
 
-
 class PurchaseOrder(models.Model):
     _inherit = 'purchase.order'
 
+    dest_address_id = fields.Many2one('res.partner', compute="_compute_dest_address_id",
+                                      store=True, readonly=False)
+
+    show_dest_address = fields.Boolean(compute="_compute_show_dest_address", compute_sudo=True)
+
+    @api.onchange('picking_type_id')
+    def _onchange_picking_type_id(self):
+        if self.picking_type_id.default_location_dest_id.usage != 'customer' and not self.show_dest_address:
+            self.dest_address_id = False
+
+    @api.depends('picking_type_id')
+    def _compute_show_dest_address(self):
+        for order in self:
+            src_location = order.picking_type_id.default_location_src_id
+            dest_location = order.picking_type_id.default_location_dest_id
+            partner = dest_location.subcontractor_ids
+            if src_location.usage == 'supplier' and (dest_location.usage == 'customer' or partner):
+                order.show_dest_address = True
+            else:
+                order.show_dest_address = False
+
+    @api.depends('picking_type_id', 'show_dest_address')
+    def _compute_dest_address_id(self):
+        for order in self:
+            dest_location = order.picking_type_id.default_location_dest_id
+            partner = dest_location.subcontractor_ids
+            if not order.show_dest_address:
+                order.dest_address_id = False
+            elif not order.dest_address_id:
+                order.dest_address_id = partner if len(partner) == 1 else False
```

Comment:
it was like that at first, but the ORM team said that having one compute for a stored and non stored field is bad practice

Reply:
which is why I had the `compute_sudo` on there

Reply:
Do you even need `_compute_show_dest_address`? It seems like you would want the address to always show if `order.picking_type_id.default_location_dest_id.is_subcontracting_location` so you can add a related field in purchase.order for `default_location_dest_id_is_subcontracting_loc`. Then since a purchase the source location should always be 'supplier' and then you can follow the logic of the existing check and add this attribute instead:

`<attribute name="attrs">{'invisible': [('default_location_dest_id_usage', '!=', 'customer'), ('default_location_dest_id_is_subcontracting_loc', '=', False)], 'required': ['|', ('default_location_dest_id_usage', '=', 'customer'), ('default_location_dest_id_is_subcontracting_loc', '=', True)]}"`